homeassistant/components/pi_hole/config_flow.py

```diff
@@ -55,33 +61,42 @@ async def async_step_init(self, user_input, is_import=False):
             location = user_input[CONF_LOCATION]
             tls = user_input[CONF_SSL]
             verify_tls = user_input[CONF_VERIFY_SSL]
-            api_token = user_input.get(CONF_API_KEY)
             endpoint = f"{host}/{location}"
 
             if await self._async_endpoint_existed(endpoint):
                 return self.async_abort(reason="already_configured")
 
             try:
-                await self._async_try_connect(
-                    host, location, tls, verify_tls, api_token
-                )
-                return self.async_create_entry(
-                    title=name,
-                    data={
-                        CONF_HOST: host,
-                        CONF_NAME: name,
-                        CONF_LOCATION: location,
-                        CONF_SSL: tls,
-                        CONF_VERIFY_SSL: verify_tls,
-                        CONF_API_KEY: api_token,
-                    },
-                )
+                await self._async_try_connect(host, location, tls, verify_tls)
             except HoleError as ex:
                 _LOGGER.debug("Connection failed: %s", ex)
                 if is_import:
                     _LOGGER.error("Failed to import: %s", ex)
                     return self.async_abort(reason="cannot_connect")
                 errors["base"] = "cannot_connect"
+            else:
+                self._config = {
+                    CONF_HOST: host,
+                    CONF_NAME: name,
+                    CONF_LOCATION: location,
+                    CONF_SSL: tls,
+                    CONF_VERIFY_SSL: verify_tls,
+                }
+                if is_import:
+                    api_key = user_input.get(CONF_API_KEY)
+                    return self.async_create_entry(
+                        title=name,
+                        data={
+                            **self._config,
+                            CONF_STATISTICS_ONLY: api_key is None,
+                            CONF_API_KEY: api_key,
+                        },
+                    )
+                statistics_only = user_input[CONF_STATISTICS_ONLY]
+                self._config[CONF_STATISTICS_ONLY] = statistics_only
+                if statistics_only:
```

Comment:
probably you can do without `statistics_only`, it's only used in these 3 lines and you can use either dictionary in the `if`